### PR TITLE
Speedy Cosmetic cleanup

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
* change `trait` to `abstract class`
* add missing `override`
* cleanup scope restrictions
* add missing `final`
* add missing `extends Product with Serializable`

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala [hunk 1]
```diff
@@ -100,31 +100,31 @@ private[lf] object Speedy {
 
   private type Actuals = util.ArrayList[SValue]
 
-  private[lf] sealed trait LedgerMode
+  sealed abstract class LedgerMode
 
-  private[lf] case class SKeyWithMaintainers(key: SValue, maintainers: Set[Party]) {
+  case class SKeyWithMaintainers(key: SValue, maintainers: Set[Party]) {
```
Comment (carlpulley-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:105]: Should we also make this `final`?

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/NormalizeRollbacks.scala [hunk 2]
```diff
@@ -90,7 +90,7 @@ private[lf] object NormalizeRollbacks {
 
   //   rule #2/#3 overlap: ROLL [ ROLL [ xs… ] ] -> ROLL [ xs… ]
 
-  private[this] def makeRoll[R](
+  private def makeRoll[R](
```
Comment (carlpulley-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/NormalizeRollbacks.scala:93]: Why do we remove `this` here?

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/NormalizeRollbacks.scala [hunk 3]
```diff
@@ -209,7 +209,9 @@ private[lf] object NormalizeRollbacks {
     }
   }
 
-  private def pushNorm[R](s: State, x: Norm)(k: (State, NodeId) => Trampoline[R]): Trampoline[R] = {
+  private def pushNorm[R](s: State, x: Norm)(
```
Comment (carlpulley-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/NormalizeRollbacks.scala:212]: Based on simple minded pattern matching, should this also be `private[this]`?

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/NormalizeRollbacks.scala [hunk 4]
```diff
@@ -134,7 +134,7 @@ private[lf] object NormalizeRollbacks {
 
   // There is no connection between the ids in the original and normalized transaction.
 
-  private[this] case class State(
+  private case class State(
```
Comment (carlpulley-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/NormalizeRollbacks.scala:137]: Should we make this `final`?

**Review by carlpulley-da — APPROVED** [review]
Some potential small niggles, but ow LGTM.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/NormalizeRollbacks.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/NormalizeRollbacks.scala:93]: reverted

**Comment by remyhaemmerle-da** [comment]
@carlpulley-da I applies the change to suggested and add 
- missing `final` in from of case classes,
- add missing `extends Product with Serializable`

**Pull request merged**